### PR TITLE
[#2976] Add length 2d land app

### PR DESCRIPTION
**Pull request by LePetitTim**

### CODE REVIEW

**Comment by LePetitTim** [comment]
We should do it with length_2d only on all apps and not only on land. Waiting for work on performance.

**Comment by LePetitTim** [comment]
We have to rebase and use new data tables

File: geotrek/api/mobile/serializers/trekking.py [hunk 1]
```diff
@@ -34,7 +33,7 @@ class Meta:
     class TrekBaseSerializer(geo_serializers.GeoFeatureModelSerializer):
         cities = serializers.SerializerMethodField()
         districts = serializers.SerializerMethodField()
-        length = serializers.SerializerMethodField()
+        length = serializers.IntegerField(source='length_2d')
```
Comment (submarcos) [inline geotrek/api/mobile/serializers/trekking.py:36]:
```suggestion
        length = serializers.FloatField(source='length_2d')
```

File: geotrek/api/v2/serializers.py [hunk 2]
```diff
@@ -502,12 +501,9 @@ class Meta:
     class PathSerializer(DynamicFieldsMixin, serializers.ModelSerializer):
         url = HyperlinkedIdentityField(view_name='apiv2:trek-detail')
         geometry = geo_serializers.GeometryField(read_only=True, source="geom3d_transformed", precision=7)
-        length_2d = serializers.SerializerMethodField()
+        length_2d = serializers.IntegerField()
```
Comment (submarcos) [inline geotrek/api/v2/serializers.py:504]:
```suggestion
        length_2d = serializers.FloatField()
```

File: geotrek/api/v2/serializers.py [hunk 3]
```diff
@@ -524,7 +520,7 @@ class TrekSerializer(PDFSerializerMixin, DynamicFieldsMixin, serializers.ModelSe
         url = HyperlinkedIdentityField(view_name='apiv2:trek-detail')
         published = serializers.SerializerMethodField()
         geometry = geo_serializers.GeometryField(read_only=True, source="geom3d_transformed", precision=7)
-        length_2d = serializers.SerializerMethodField()
+        length_2d = serializers.IntegerField()
```
Comment (submarcos) [inline geotrek/api/v2/serializers.py:523]:
```suggestion
        length_2d = serializers.FloatField()
```

File: geotrek/core/models.py [hunk 4]
```diff
@@ -42,7 +43,8 @@ class PathManager(models.Manager):
     def get_queryset(self):
         """Hide all ``Path`` records that are not marked as visible.
         """
-        return super().get_queryset().filter(visible=True).annotate(length_2d=Length('geom'))
+        qs = super().get_queryset().filter(visible=True).annotate(length_2d=Length('geom'))
+        return qs
```
Comment (submarcos) [inline geotrek/core/models.py:47]: why ?

File: geotrek/core/tests/test_topology.py [hunk 5]
```diff
@@ -62,7 +62,7 @@ def test_length(self):
     def test_length_2d(self):
         e = TopologyFactory.build()
         e.save()
-        self.assertEqual(e.length_2d, None)
+        self.assertEqual(Topology.objects.get().length_2d, 0)
```
Comment (submarcos) [inline geotrek/core/tests/test_topology.py:65]: e.refresh_from_db should be better to be sure testing right instance

File: docs/changelog.rst [hunk 6]
```diff
@@ -76,7 +82,6 @@ CHANGELOG
 - Add image widget to tinymce editors by default
 - Delete filenames in captions of attachments when importing from Apidae (#2698)
 - Add copyright when importing from Apidae on attachments (#2698)
-- Improve basic fixture for Feedback app allowing to initialize Report form in one go
```
Comment (submarcos) [inline docs/changelog.rst:79]: don't delete this entry

File: geotrek/core/models.py [hunk 7]
```diff
@@ -380,6 +381,14 @@ def distance(self, to_cls):
         return None
 
 
+class TopologyManager(NoDeleteManager):
```
Comment (submarcos) [inline geotrek/core/models.py:384]: use a base Manager

**Pull request merged**